### PR TITLE
Sprint 3 Ticket

### PR DESCRIPTION
**Pull request by vikram-kher**
----------------------------------
I could not get these two changes to work:

filterByFormula: {ID}=${userID}
filterByFormula: {Code}=${companyCode}

It says that the formula is invalid.
----------------------------------
Sorry for being late!

Ticket Details:
-----------------
1. Completed the user route
    - Returns user's Full Name, Email, Current Title, Answered Questions 
2. Completed the company route
    - Returns company ID and company name
3. Created tests for the company route
4. Created a single test for the user route

### CODE REVIEW

File: server/data_access_layer/company.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+const { extractContentFromRecords, getAllFromTable } = require('./helpers');
+
+async function getCompany(companyCode) {
+  const companyTable = base('Company Codes').select({
+    // Filter by company Code
+    filterByFormula: '{Code}=' + companyCode,
+    view: 'Grid view'
+  });
+  const company = await getAllFromTable(companyTable);
+  var companyData = { 'ID': company[0]['fields']['ID'], 'Title': company[0]['fields']['Title'] }
```
Comment (chuanxiuXiong) [inline server/data_access_layer/company.js]: optional:

```
const company = await getAllFromTable(companyTable)[0];
const companyData = { ID: company.fields.ID, Title: company.fields.Title };
return companyData;
```

File: server/data_access_layer/user.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    // Filter by user id
+    filterByFormula: '{ID}=' + userID,
+    view: 'Grid view'
+  });
+
+  const user = await getAllFromTable(userTable);
+  var userData = {
+    'Full Name': user[0]['fields']['Full Name'], 'Email': user[0]['fields']['Email'],
+    'Current Title': user[0]['fields']['Current Title'], 'Answered': user[0]['fields']['Answered']
+  };
```
Comment (chuanxiuXiong) [inline server/data_access_layer/user.js]: optional:

```
const user = await getAllFromTable(userTable)[0];
const userData = {
     'Full Name': user['fields']['Full Name'], 'Email': user['fields']['Email'],
      'Current Title': user['fields']['Current Title'], 'Answered': user['fields']['Answered']
    };
```

File: server/data_access_layer/user.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
+const { base } = require('./index');
+const { extractContentFromRecords, getAllFromTable } = require('./helpers');
+
+async function getUser(userID) {
+  const userTable = base('Users').select({
+    // Filter by user id
+    filterByFormula: '{ID}=' + userID,
```
Comment (chuanxiuXiong) [inline server/data_access_layer/user.js]: optional: you could use string literal, which is more common in JS
```
filterByFormula: `{ID}=${userID}`,
```

File: server/data_access_layer/user.js (hunk 2 above)
Comment (apetranik) [reply server/data_access_layer/user.js]: agreed ^ 
+ can you also split each field onto a newline so it's easier to read

File: server/data_access_layer/company.js (hunk 1 above)
Comment (apetranik) [reply server/data_access_layer/company.js]: agreed ^ 
+ can you also split each field onto a newline so it's easier to read

File: server/data_access_layer/company.js [hunk 4]
```diff
@@ -0,0 +1,17 @@
+const { base } = require('./index');
+const { extractContentFromRecords, getAllFromTable } = require('./helpers');
+
+async function getCompany(companyCode) {
+  const companyTable = base('Company Codes').select({
+    // Filter by company Code
+    filterByFormula: '{Code}=' + companyCode,
```
Comment (apetranik) [inline server/data_access_layer/company.js]: filterByFormula: `{Code}=${companyCode}`,

File: server/api/routes/company.js [hunk 5]
```diff
@@ -0,0 +1,19 @@
+const { getCompany } = require('../../data_access_layer/company');
+
+module.exports = app => {
+  app.get('/api/companycode/:code', async (req, res) => {
+    try {
+      const { code } = req.params;
+      const company = await getCompany(code);
+      return res.send(company).status(200);
```
[the remark this answers is not in the document]
Comment (apetranik) [reply server/api/routes/company.js]: good catch!

File: __tests__/server/company-routes.spec.js [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+    var keys = Object.keys(res.body);
+    expect(keys[0]).toMatch('ID');
+    expect(keys[1]).toMatch('Title');
+    expect(res.status).toBe(200);
+  });
+});
+
+describe('Checks to see if invalid url is handled', () => {
+  it('should return a 404 error', async () => {
+    const res = await request.get('/api/companycod/1111');
```
Comment (apetranik) [inline __tests__/server/company-routes.spec.js]: caught this companycode** - I'll fix it

**Pull request merged**